### PR TITLE
Standardize namespace and name normalization to match Python library

### PR DESCRIPTION
**Pull request by brbayes-msft**
This change adjusts the normalization for namespace and names to only apply changes that are necessary to be compliant for the given package protocol.  This aligns with how the Python library implements normalization, and is done per the request in this pull request (https://github.com/package-url/packageurl-dotnet/pull/19#discussion_r1007339775).

### CODE REVIEW

**Comment by brbayes-msft** [comment]
Hi @am11.  Could you please take a look at these changes?  This unblocks some use of Maven packages that are not all lower case.

**Review by pmalmsten — DISMISSED** [review]

File: src/PackageUrl.cs [hunk 1]
```diff
@@ -263,8 +263,8 @@ private string ValidateNamespace(string @namespace)
             }
             return Type switch
             {
-                "vsm" or "cran" => WebUtility.UrlDecode(@namespace),
-                _ => WebUtility.UrlDecode(@namespace.ToLower())
+                "bitbucket" or "github" or "pypi" or "gitlab" => WebUtility.UrlDecode(@namespace.ToLower()),
```
Comment (am11) [inline src/PackageUrl.cs:266]: Looks like we don't have a test case for github and gitlab which exercises this branch.

**Review by am11 — APPROVED** [review]
LGTM, thanks!

File: src/PackageUrl.cs (hunk 1 above)
Comment (brbayes-msft) [reply src/PackageUrl.cs:266]: I added a new test case for github.  I did not see a good example for in the PURL types document (https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst), or in other test suite data files for gitlab, and I didn't want to add in a bad example that got copied by others.

**Review by am11 — APPROVED** [review]

**Pull request merged**